Activities/Cryptography/UiPath.Cryptography.Activities/NetCore/ViewModels/KeyedHashFileViewModel.cs

```diff
@@ -89,54 +89,70 @@ protected override void InitializeModel()
             KeySecureString.IsVisible = false;
             KeySecureString.OrderIndex = propertyOrderIndex++;
 
-            KeyInputModeSwitch.IsVisible = false;
+            var useKeyMenuAction = new MenuAction
+            {
+                DisplayName = Resources.Activity_KeyedHashFile_Property_Key_SwitchName,
+                Handler = UseKeyHandler
+            };
+
+            var useKeySecureStringMenuAction = new MenuAction
+            {
+                DisplayName = Resources.Activity_KeyedHashFile_Property_KeySecureString_SwitchName,
+                Handler = UseKeySecureStringHandler
+            };
+
+            _useKeySecureString = KeySecureString.Value != null;
```

Comment:
Using this private member to keep track of state based on the value of a given property initially, but THEN updating it via menu handlers feels wrong and is error prone.

Why not just keep the dedicated input mode property? It would make the code a whole lot cleaner and easier to track/understand.

It's also a pattern we have on other activities, where the mode is explicitly captured in the activity state/XAML.

You can also use this information to only show the validation errors for the current mode, e,g. no need to generate/show a validation error for secure string if we're not in secure string mode - and this works in both the CacheMetadata and ViewModel layers.